app/src/main/java/org/fossasia/openevent/app/core/auth/login/LoginFragment.java

```diff
@@ -50,24 +56,43 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     public void onStart() {
         super.onStart();
         getPresenter().attach(this);
-        binding.setLogin(getPresenter().getLogin());
-        getPresenter().start();
+
+        //attaching the view model
+        loginFragmentViewModel = ViewModelProviders.of(this, viewModelFactory).get(LoginViewModel.class);
+        handleEmails();
+
+        //getPresenter().start();
+        binding.setLogin(loginFragmentViewModel.getLogin());
+
         binding.btnLogin.setOnClickListener(view -> {
             if (!validator.validate())
                 return;
 
             ViewUtils.hideKeyboard(view);
 
             String url = binding.url.baseUrl.getText().toString().trim();
-            getPresenter().setBaseUrl(url, binding.url.overrideUrl.isChecked());
-            getPresenter().login();
+            loginFragmentViewModel.setBaseUrl(url, binding.url.overrideUrl.isChecked());
+            loginFragmentViewModel.login().observe(this, this::onResponse);
         });
 
         binding.signUpLink.setOnClickListener(view -> openSignUpPage());
-
         binding.forgotPasswordLink.setOnClickListener(view -> openForgotPasswordPage());
     }
 
+    private void handleEmails() {
+        emailsList = loginFragmentViewModel.getEmailList();
+        if(emailsList != null)
```

Comment:
Rather than proactively fetching, expose a live data instead. Also, only post if not null so that views don't have to apply logic